Client/game_sa/CPoolsSA.cpp

```diff
@@ -329,6 +332,105 @@ void CPoolsSA::DeleteAllObjects()
     }
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+//                                     BUILDINGS POOL                                   //
+//////////////////////////////////////////////////////////////////////////////////////////
+
+inline bool CPoolsSA::AddBuildingToPool(CClientBuilding* pClientBuilding, CBuildingSA* pBuilding)
+{
+    // Grab the new object interface
+    CBuildingSAInterface* pInterface = pBuilding->GetBuildingInterface();
+
+    if (!pInterface)
+        return false;
+
+    DWORD dwElementIndexInPool = GetBuildingPoolIndex((std::uint8_t*)pInterface);
+    if (dwElementIndexInPool >= MAX_BUILDINGS)
+    {
+        return false;
+    }
+
+    m_buildingPool.arrayOfClientEntities[dwElementIndexInPool] = {pBuilding, (CClientEntity*)pClientBuilding};
+
+    // Increase the count of objects
+    ++m_buildingPool.ulCount;
+
+    return true;
+}
+
+CBuilding* CPoolsSA::AddBuilding(CClientBuilding* pClientBuilding, uint16_t modelId, CVector *vPos, CVector4D *vRot, uint8_t interior)
+{
+    if (!HasFreeBuildingSlot())
+        return nullptr;
+
+    // Load building
+    SFileObjectInstance instance;
+    instance.modelID = modelId;
+    instance.lod = -1;
+    instance.interiorID = interior;
+    instance.position = *vPos;
+    instance.rotation = *vRot;
+
+    // Fix strange SA rotation 
+    instance.rotation.fW = -instance.rotation.fW;
+
+    CFileLoaderSA loader{};
+    auto pBuilding = static_cast<CBuildingSAInterface*>(loader.LoadObjectInstance(&instance));
```

Comment:
What is the reason for using a non-static LoadObjectInstance method? Isn't it better to make it static and just CFileLoaderSA::LoadObjectInstance(&instance)? From what I can see, this method is static in SA(see reversed SA or IDA).

Reply:
For some architectural reasons i prefer loader classes over functions.
I can make it simpler with using static function, because we don't have DI / tests for this code

Reply:
This is OK. But I prefer to keep the code as close to SA as possible.